tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -332,7 +374,151 @@ TEST_F(OverlapTest, ReduceScatterBasedPipeliningHostIrImplementation) {
 
     hie.runWithInput(std::move(inputs));
   }
-  validate();
 }
 
+using RingBasedOverlapTestParams = std::tuple<
+    bool, // use_coalescence
```

Comment:
For this particular ring-based implementation, isn't there a strong preference to use coalescing? If so, I'd rather remove the code for not coalescing and simplify.

Reply:
> For this particular ring-based implementation, isn't there a strong preference to use coalescing? If so, I'd rather remove the code for not coalescing and simplify.

In theory, yes, it is correct. However, ProcessGroupUCC doesn't implement coalescing, so we need an alternative for this backend, hence the branching. UCC has great advantages for P2P, e.g., regarding SM occupation, that is why it is important to support this backend eventually.

Reply:
would you like me to remove it in the meantime, and add it separately when we add support to UCC ?

Reply:
> would you like me to remove it in the meantime, and add it separately when we add UCC support ?

Yes, that sounds the best to me.

Reply:
done